docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
 Before you begin this tutorial, complete the [Extending an existing plug-in](cli-extending-a-plugin.md) tutorial.
 
 ## Overview
-This tutorial demonstrates how to create a brand new Zowe&trade; CLI plug-in that uses Zowe CLI Node.js programmatic APIs.
+The advantage of Zowe CLI and of the CLI approach in mainframe development is that it allows for combining different developer tools for new and interesting uses.
+
+This tutorial demonstrates how to create a brand new Zowe CLI plug-in that uses Node.js to create a client-side API.
+
+After following all the steps, you will have created a data set diff utility plug-in called **Files Util Plug-in**. The plug-in provides an output that can be sent to a third-party utility to display easy-to-read side-by-side diffs of data set member contents, as shown in the image at the [bottom of this page](../extend-cli/cli-developing-a-plugin#bringing-together-new-tools).
+
+The completed source for this tutorial can be found on the `develop-a-plugin` branch of the [zowe-cli-sample-plugin repository](https://github.com/zowe/zowe-cli-sample-plugin).
+
+**Note:** If you are ready to create your own unique Zowe CLI plug-in, refer to the notes below each tutorial step for guidance.
 
-At the end of this tutorial, you will have created a data set diff utility plug-in for Zowe CLI, from which you can pipe
-your plugin's output to a third-party utility for a side-by-side diff of data set member contents.
+## Setting up the new sample plug-in project
+ Download the sample plug-in source and delete the irrelevant content to set up your plug-in project.
+ 
+ Follow these steps:
 
-![Side by Side Diff](../../images/guides/CLI/htmlDiff.png)
+1. Open a terminal and run the command `mkdir zowe-tutorial`.
+    
+    **Note:** All the files created through this tutorial are saved in this tutorial directory.
+2. Enter `cd zowe-tutorial` to change directory into your `zowe-tutorial` folder.
+3. Download the [source code zip file](https://github.com/zowe/zowe-cli-sample-plugin/archive/refs/heads/master.zip) from the Zowe CLI sample plug-in repository.
+4. In your File Explorer, extract the zip file to the  `zowe-tutorial` folder.
+5. Rename the `zowe-cli-sample-plugin-master` directory to `files-util`.
+    
+    This is the project directory used throughout the rest of this tutorial. 
+6. Delete all content within the following folders:
+    - `src/api`
+    - `src/cli`
```

Comment:
two things:
1. should we mention that they should delete the `src/overrides` folder  OR are we going to delete it eventually from the master branch?
2. We should add a not to call out the `src/imperative.ts`. 
That way people know not to delete that one file since it will be used later in the doc.

Reply:
For Item 1: We added a note linking to [Zowe CLI secrets for kubernetes ](https://github.com/zowe/zowe-cli-secrets-for-kubernetes)repository. Dev team will remove this from the CLI repository.